compiler/rustc_trait_selection/src/traits/const_evaluatable.rs

```diff
@@ -108,6 +108,9 @@ pub fn is_const_evaluatable<'cx, 'tcx>(
     }
 
     let future_compat_lint = || {
+        if tcx.features().generic_const_exprs {
+            return;
+        }
```

Comment:
We already have a branch further down which checks for `generic_const_exprs`. Can you merge these two instead? Also please move the definition of `future_compat_lint` into the if statement

https://github.com/rust-lang/rust/blob/50872bdb99c96716ee50a3b9613c395302b99767/compiler/rustc_trait_selection/src/traits/const_evaluatable.rs#L134-L171

Reply:
I don't want to see a way to join both branches, since they test pretty different things, but I've merged them all into one large match statement, which I guess is more efficient (if this was the goal). I've moved the defn into the if block.